# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/RemoteDecoderChild.cpp
# Commit: 3dd1e88edaf1
# Full Hash: 3dd1e88edaf16d10ac618dfb2b46193ec1ec0a62
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2020-10-21 09:51:28
# Regressor Bug: 1595994
# File Overlap Count: 1
# Description:
#   Bug 1595994 - P24. Always initialize Remote Decoder Manager thread early. r=mattwoodrow
#   
#   Some cleanup on how we create the RemoteDecoderChild object, simplifying and clarifying.
#   
#   Depends on D92362
# ==============================================================================

diff -r 084a23683a4e -r 3dd1e88edaf1 dom/media/ipc/RemoteDecoderChild.cpp
--- a/dom/media/ipc/RemoteDecoderChild.cpp	Tue Oct 20 23:30:58 2020 +0000
+++ b/dom/media/ipc/RemoteDecoderChild.cpp	Tue Oct 20 23:31:11 2020 +0000
@@ -11,8 +11,13 @@
 
 RemoteDecoderChild::RemoteDecoderChild(bool aRecreatedOnCrash)
     : ShmemRecycleAllocator(this),
-      mThread(RemoteDecoderManagerChild::GetManagerThread()),
-      mRecreatedOnCrash(aRecreatedOnCrash) {}
+      mThread(GetCurrentSerialEventTarget()),
+      mRecreatedOnCrash(aRecreatedOnCrash) {
+  MOZ_DIAGNOSTIC_ASSERT(
+      RemoteDecoderManagerChild::GetManagerThread() &&
+          RemoteDecoderManagerChild::GetManagerThread()->IsOnCurrentThread(),
+      "Must be created on the manager thread");
+}
 
 void RemoteDecoderChild::HandleRejectionError(
     const ipc::ResponseRejectReason& aReason,